pkg/utils/slice.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"cmp"
+	"slices"
+)
+
+func DedupeSlice[T cmp.Ordered](s []T) []T {
+	if len(s) < 2 {
+		return s
```

Comment:
We can use this instead: https://pkg.go.dev/slices#Compact

Reply:
Nice, that's what I was looking for in that package, but I kept getting stuck at looking for the word dedupe. Thank you 🙏 

This does not say if the input has to be sorted. Guess it has to be. Will try it out and switch it out.

Reply:
Yeah, sorting would be needed. It's only looking at adjacent elements:

https://cs.opensource.google/go/go/+/refs/tags/go1.24.2:src/slices/slices.go;l=363-382

Reply:
hmm, it does not change length of the slice

Reply:
oh, the code does seem to, but trying a payground and it shows the full length still - https://go.dev/play/p/nXrvl-5XZCv, I must be doing something wrong

Reply:
🤦 of course :-)